data_model/src/domain.rs

```diff
@@ -108,6 +127,42 @@ impl Domain {
             logo: None,
         }
     }
+
+    /// Returns a reference to the account corresponding to the account id.
+    pub fn get_account(&self, account_id: &<Account as Identifiable>::Id) -> Option<&Account> {
+        self.accounts.get(account_id)
+    }
+
+    /// Returns a mutable reference to the account corresponding to the account id.
+    pub fn get_account_mut(
+        &mut self,
+        account_id: &<Account as Identifiable>::Id,
+    ) -> Option<&mut Account> {
+        self.accounts.get_mut(account_id)
+    }
+
+    /// Adds account into the domain
+    pub fn add_account(&mut self, account: Account) -> Option<Account> {
+        self.accounts.insert(account.id.clone(), account)
+    }
+
+    /// Removes account from the domain
+    pub fn remove_account(
+        &mut self,
+        account_id: &<Account as Identifiable>::Id,
+    ) -> Option<Account> {
+        self.accounts.remove(account_id)
+    }
+
+    /// Gets an iterator over accounts of the domain
+    pub fn accounts(&self) -> impl Iterator<Item = &Account> {
```

Comment:
if we wanted to expose iterators over FFI, we would have to make a special wrapper around the iterator

Reply:
or return a vec here like:
```
    pub fn accounts(&self) -> Vec<&Account>
```

Reply:
this is totally fine and an API can be made to mirror this for FFI